week1/exercise1.hs

```diff
@@ -2,6 +2,7 @@ toDigits :: Integer -> [Integer]
 toDigits n
     | n <= 0    = []
     | otherwise = map (\a -> read [a] :: Integer) (show n)
+    -- alternative way
     -- | otherwise = toDigits (div n 10) ++ [mod n 10]
 
 toDigitsRev :: Integer -> [Integer]
```

Comment:
you can use `.` to compose functions, and that way you don't need to specify the argument
```
toDigitsRev = reverse . toDigits
```

Reply:
you can also use this syntax: 

```
doubleEveryOther x =
     [res | (index, value) <- xs , let res = value * (mod index 2 + 1)]
  where
     enumerate :: [Integer] -> [(Integer, Integer)]
     enumerate [] = []
     enumerate xs = zip [0..] xs
```

Reply:
For `sumDigits` you can play around with `concatMap` and `sum` like we discussed earlier

Reply:
you can actually rewrite this as function composition with a bit of rewriting:

```
validate :: Integer -> Bool
validate = (0 ==) . (`mod` 10) . sumDigits . doubleEveryOther . toDigitsRev
```